src/main/java/com/rackspace/salus/common/web/AbstractRestExceptionHandler.java

```diff
@@ -179,4 +180,16 @@ protected void logRequestFailure(HttpServletRequest request, Exception e) {
     return respondWith(request, HttpStatus.SERVICE_UNAVAILABLE, ResponseMessages.kafkaExceptionMessage);
   }
 
+  @ExceptionHandler(MethodArgumentNotValidException.class)
+  public ResponseEntity<?> handleMethodArgumentNotValidException(HttpServletRequest request, MethodArgumentNotValidException e) {
+    logRequestFailure(request, e);
+    StringBuilder message = new StringBuilder();
+    e.getBindingResult().getFieldErrors().stream().forEach(error -> {
+      message.append("Validation failed for [field]: ").append(error.getField())
+          .append(" with [message]: ").append(error.getDefaultMessage()).append(", ");
+    });
+    message.delete(message.length() - 2, message.length());
```

Comment:
There's a handy [joining](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/stream/Collectors.html#joining(java.lang.CharSequence)) collector that will avoid the need to manually append and trim the StringBuilder. I'm also providing a suggestion of a further trimmed down wording that should read pretty nicely for most validation messages.

```suggestion
    String message = "One or more field validations failed: " +
    e.getBindingResult().getFieldErrors().stream()
    .map(error -> error.getField() + " " + error.getDefaultMessage())
    .collect(Collectors.joining(", ");
```

Using the example from Slack it should produce:
```
One or more field validations failed: resourceId must match \"[A-Za-z0-9.:-]+\", presenceMonitoringEnabled must not be null
```

Reply:
It is giving nice response but shouldn't we have a specific marker on request body field that is failing. 
For example:-
**One or more field validations failed: provider may not be empty, name zone name must be public** 
Here **name zone name must be public** is kind of looking weird or maybe i can change the default message in code to **must be public** only.
What's your thought on this?

Reply:
Yeah, I'm thinking cases where the wording becomes awkward it's because the validation message wasn't phrased correctly. So, yes, in this case the annotation should have said "must be public".

I guess that means it would be good if you could skim the validation annotations to make sure they start with "must" or "may" and tend not to mention the field.

Reply:
As for

> shouldn't we have a specific marker on request body field that is failing

that's good point. Maybe something simple like putting single quotes around the field name part.